### PR TITLE
Feat/profile cover pic

### PR DESCRIPTION
**Pull request by Hakdoff**
OVerview
Added profile picture in navbar

Snippet
image

### CODE REVIEW

**Comment by mikco-hipe** [comment]
Please change base branch to dev

File: src/database/migrations/0001_01_01_000000_create_users_table.php [hunk 1]
```diff
@@ -46,4 +47,4 @@ public function down(): void
         Schema::dropIfExists('password_reset_tokens');
         Schema::dropIfExists('sessions');
     }
-};
+};
```
Comment (hipe01mark) [inline src/database/migrations/0001_01_01_000000_create_users_table.php]: Please check!

File: src/database/migrations/2024_04_04_073207_create_profiles_table.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    }
+
+    /**
+     * Reverse the migrations.
+     */
+    public function down(): void
+    {
+        Schema::dropIfExists('profiles');
+    }
+};
```
Comment (hipe01mark) [inline src/database/migrations/2024_04_04_073207_create_profiles_table.php]: Please check!

File: src/resources/scss/auth/login.scss [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    &__padding4 {
+        @apply py-4;
+    }
+
+    &__padding5 {
+        @apply py-5;
+    }
+}
```
Comment (hipe01mark) [inline src/resources/scss/auth/login.scss]: Please check!

File: src/resources/scss/auth/register.scss [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    }
+
+    &_cancelButton {
+        @apply w-full px-4 py-2 mr-4 text-sm font-medium text-warmbeige bg-softblack bg-opacity-80 rounded-full border border-gray-200 hover:bg-red-700 focus:outline-none focus:ring-4 focus:ring-red-300;
+    }
+
+    &_saveButton {
+        @apply w-full px-4 py-2 text-sm font-medium text-warmbeige bg-softblack rounded-full border border-transparent hover:bg-green-700 focus:outline-none focus:ring-4 focus:ring-green-300;
+    }
+}
```
Comment (hipe01mark) [inline src/resources/scss/auth/register.scss]: Please check!

File: src/resources/scss/dashboard/dashboard.scss [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+
+        to {
+            transform: rotate(360deg);
+        }
+    }
+
+    .top-negative {
+        top: -155px;
+    }
+}
```
Comment (hipe01mark) [inline src/resources/scss/dashboard/dashboard.scss]: Please check!

File: src/resources/scss/profile/profile.scss [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+        to {
+            transform: rotate(360deg);
+        }
+    }
+
+    .top-negative {
+        top: -155px;
+    }
+}
```
Comment (hipe01mark) [inline src/resources/scss/profile/profile.scss]: Please check!

File: src/resources/views/pages/dashboard/like-button.blade.php [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+@else
+  <form action="{{route('like.post', $post->id)}}" method="POST">
+    @csrf
+@endif
+    <button type="submit" class="d-dashboard__like-comm">
+    <span class="flex justify-center items-center gap-2"> {{$post->likes()->distinct('user_id')->count()}}
+    @include('svg.like')
+    </button>
+  </form>
+@endauth
```
Comment (hipe01mark) [inline src/resources/views/pages/dashboard/like-button.blade.php]: Please check!

File: src/resources/scss/auth/login.scss (hunk 3 above)
Comment (Hakdoff) [reply src/resources/scss/auth/login.scss]: Sir, peter will be the who will add break lines on the blade and scss files. As my visual studio automatically remove the break lines once I save the file.

**Comment by PeterDMadrid** [comment]
Added line breaks. Please check 🐐

File: src/resources/views/pages/dashboard/post.blade.php [hunk 8]
```diff
@@ -16,4 +21,4 @@
             </div>
         </div>
     </form>
-</body>
+</body>
```
Comment (hipe01mark) [inline src/resources/views/pages/dashboard/post.blade.php]: Please check!

File: src/app/Models/Profile.php [hunk 9]
```diff
@@ -28,4 +30,28 @@ public function user(): BelongsTo
...
+    /**
+     * Get image URL
+     */
+    public function getImageURL(): string
+    {
+        if($this->profilepic){
+            $cleanedPath = str_replace('public/', '', $this->profilepic);
+            return url('storage/'. $cleanedPath);
+        }
+        return "https://images.unsplash.com/photo-1438761681033-6461ffad8d80?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=400&fit=max&ixid=eyJhcHBfaWQiOjE0NTg5fQ{{$this->name}}";
```
Comment (hipe01mark) [inline src/app/Models/Profile.php]: is it possible to make a const variable and then concatenate it?

File: src/app/Models/Profile.php [hunk 10]
```diff
@@ -28,4 +30,28 @@ public function user(): BelongsTo
...
+    /**
+     * Get cover URL
+     */
+    public function getCoverURL(): string
+    {
+        if($this->coverpic){
+            $cleanedPath = str_replace('public/', '', $this->coverpic);
+            return url('storage/'. $cleanedPath);
+        }
+        return "https://images.unsplash.com/photo-1438761681033-6461ffad8d80?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=400&fit=max&ixid=eyJhcHBfaWQiOjE0NTg5fQ{{$this->name}}";
```
Comment (hipe01mark) [inline src/app/Models/Profile.php]: is it possible to make a const variable and then concatenate it?

File: src/resources/views/pages/dashboard/navbar.blade.php [hunk 11]
```diff
@@ -1,48 +1,82 @@
...
+                                <path
+                                    d="M25,18.4c11,0 20,-2.4 20,-5.4c0,-1.1 -1.2,-2.1 -3.2,-2.9c0.8,0.5 1.2,1.1 1.2,1.7c0,2.6 -8.1,4.7 -18,4.7c-9.9,0 -18,-2.2 -18,-4.8c0,-0.6 0.4,-1.2 1.2,-1.7c-2,0.8 -3.2,1.9 -3.2,2.9c0,3 9,5.5 20,5.5z">
+                                </path>
+                                <path
+                                    d="M25,21c-8.8,0 -16.3,-1.6 -19.9,-4h-0.1c0,2.3 0.8,4.7 1.9,7.1c-5.9,4.1 -4.8,14.9 5.1,12.9c0.5,-0.1 0.5,-0.5 0,-0.6c-3.9,-0.5 -7.9,-5.3 -4.1,-10.3c2.3,4.4 5.5,8.8 7.2,12.9v0c0.7,1.7 4.8,3 9.9,3c5.1,0 9.2,-1.3 9.9,-3v0c3,-7 10.1,-15 10.1,-22h-0.1c-3.6,2.4 -11.1,4 -19.9,4zM18,9c0,0 -4.9,-0.4 -4.9,2.2c0,1.1 0.8,2.6 4.2,2.6c3,0 6.8,-3.7 11.9,-3.7c3.7,0 3.7,2.5 0.7,2.5c-2,0 -3,-0.8 -3,-0.8c-0.6,-0.4 -1.3,0.4 -1,1.1c0,0 0,1.1 4,1.1c4,0 4.6,-1.9 4.5,-2.6c-0.1,-1.4 -1.4,-2.7 -3.5,-3.1c-1.9,-0.3 -4,-0.5 -7,0.6c-3,1.1 -5.1,3.1 -7.1,3.1c-2,0 -3.6,-2.3 1.2,-2c2.3,0.2 1,-1 0,-1z">
+                                </path>
+                            </g>
+                        </g>
+                    </svg><a href="{{ route('dashboard', ['id' => auth()->id()]) }}">Microblog
```
Comment (hipe01mark) [inline src/resources/views/pages/dashboard/navbar.blade.php]: Create a seperate file for SVG

**Comment by hipe01mark** [comment]
Fix the conflict as well.

File: src/resources/views/pages/dashboard/post.blade.php (hunk 8 above)
Comment (hipe01mark) [reply src/resources/views/pages/dashboard/post.blade.php]: one more. Check this

**Comment by hipe01mark** [comment]
Do not merge dev branch to your current branch to resolve your conflicts

Just do the fetch and rebase strategy.

Kindly drop this commit

[b1c2fe5](https://github.com/xReddskie/microblog-ojt2/pull/30/commits/b1c2fe5a87a34cf7123c27e6a1518d842920398f)


TO REMOVE THAT COMMIT

git log

count until you find the that commit
Ex. git rebase -i HEAD^10

change that commit from "pick" to "drop"
then save it

git push origin {{ your branch }} -f

**Review by hipe01mark — APPROVED** [review]
LGTM 🙇

**Pull request merged**